ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/DistributingExecutorTest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.server.computation;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.confluent.ksql.parser.KsqlParser.PreparedStatement;
+import io.confluent.ksql.parser.tree.EmptyStatement;
+import io.confluent.ksql.rest.entity.CommandStatus;
+import io.confluent.ksql.rest.entity.CommandStatus.Status;
+import io.confluent.ksql.rest.entity.CommandStatusEntity;
+import io.confluent.ksql.rest.server.computation.CommandId.Action;
+import io.confluent.ksql.rest.server.computation.CommandId.Type;
+import io.confluent.ksql.rest.server.computation.DistributingExecutor.SchemaInference;
+import io.confluent.ksql.services.ServiceContext;
+import io.confluent.ksql.util.KsqlException;
+import java.time.Duration;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DistributingExecutorTest {
+
+  private static final Duration DURATION_10_MS = Duration.ofMillis(10);
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Mock CommandQueue queue;
+  @Mock QueuedCommandStatus status;
+  @Mock ServiceContext serviceContext;
+
+  @Test
+  public void shouldEnqueueSuccessfulCommand() throws InterruptedException {
+    // Given:
+    final CommandId command = new CommandId(Type.STREAM, "stream", Action.CREATE);
+    final CommandStatus commandStatus = new CommandStatus(Status.SUCCESS, "");
+
+    when(queue.enqueueCommand(any(), any(), any())).thenReturn(status);
```

Comment:
- why not setup these mocks in a `setUp` method? 
- why not make `command` and `commandStatus` and `preparedStatement ` constants?
- how about creating an instance `new DistributingExecutor(queue, DURATION_10_MS, schemaInference)` in setup method?

Otherwise what tends to happen is people come and add more tests, cutting and pasting the same `given` code.

The idea is that `given` should be very succinct - only containing the bits pertinent to the test in question.

Also, this looks like there tests to me, (each test should, in general, test one thing):

```java
@Test
public void shouldEnqueueSuccessfulCommand() throws Exception {
   // When:
   executor.execute(preparedStatement, null, serviceContext, config, overrides);

   // Then:
   verify(queue).enqueueCommand(preparedStatement, config, overrides); <-- checking all params.
}

@Test
public void shouldInferSchemas() throws Exception {
   // Given:
   when(schemaInferrer.infer(preparedStatement).thenReturn(statementWithSchema);

   // When:
   executor.execute(preparedStatement, null, serviceContext, config, overrides);

   // Then:
   verify(queue).enqueueCommand(eq(statementwithSChema), any(), any());
}

@Test
public void shouldReturnCommandStatus() throws Exception {
    // When:
   final KsqlEntity result = executor.execute(preparedStatement, null, serviceContext, config, overrides);

   // Then:
   assertThat(result, is(new CommandStatusEntity(
        "",
        command,
        commandStatus,
        1L
    )));
}
```

By splitting the tests up it makes it much easier for others to see what each test is testing and it means if someone introduces a bug its likely only _one_ specific targeted test will fail.

The other tests you've added would benefit from similar refactoring.